cg/services/illumina_post_processing_service/illumina_post_processing_service.py

```diff
@@ -0,0 +1,105 @@
+"""Module that holds the illumina post-processing service."""
+
+import logging
+from pathlib import Path
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants.devices import DeviceType
+from cg.exc import MissingFilesError, FlowCellError
+from cg.models.flow_cell.flow_cell import FlowCellDirectoryData
+from cg.services.illumina_post_processing_service.database_utils import store_illumina_flow_cell
+from cg.services.illumina_post_processing_service.utils import (
+    create_delivery_file_in_flow_cell_directory,
+    get_flow_cell_model_from_run_parameters,
+)
+from cg.services.illumina_post_processing_service.validation import (
+    is_flow_cell_ready_for_postprocessing,
+)
+from cg.store.models import IlluminaFlowCell
+from cg.store.store import Store
+
+LOG = logging.getLogger(__name__)
+
+
+class IlluminaPostProcessingService:
+    def __init__(self, status_db: Store, housekeeper_api: HousekeeperAPI, dry_run: bool) -> None:
+        self.status_db: Store = status_db
+        self.hk_api: HousekeeperAPI = housekeeper_api
+        self.dry_run: bool = False
+
+    @staticmethod
+    def store_illumina_flow_cell(
+        flow_cell: FlowCellDirectoryData,
+        store: Store,
+    ) -> IlluminaFlowCell:
+        """
+        Create flow cell from the parsed and validated flow cell data.
+        And add the samples on the flow cell to the model.
+        """
+        model: str | None = get_flow_cell_model_from_run_parameters(flow_cell.run_parameters_path)
+        flow_cell = IlluminaFlowCell(
+            internal_id=flow_cell.id, type=DeviceType.ILLUMINA, model=model
+        )
+        new_flow_cell: IlluminaFlowCell = store.add_illumina_flow_cell(flow_cell)
+        if not flow_cell:
+            raise ValueError(f"Failed to store flow cell {flow_cell.id}")
+        return new_flow_cell
+
+    @staticmethod
+    def store_illumina_sequencing_metrics(flow_cell: IlluminaFlowCell) -> None:
+        """Store illumina run metrics in the status database."""
+        pass
+
+    @staticmethod
+    def store_illumina_sample_sequencing_metrics():
+        """Store illumina sample sequencing metrics in the status database."""
+        pass
+
+    def store_illumina_flow_cell_data(self, flow_cell: FlowCellDirectoryData) -> None:
+        """Store flow cell data in the status database."""
+        flow_cell: IlluminaFlowCell | None = self.store_illumina_flow_cell(
```

Comment:
I think adding `None` here is confusing. 
```suggestion
        flow_cell: IlluminaFlowCell = self.store_illumina_flow_cell(
```